xapian-letor/api/featuremanager.cc

```diff
@@ -69,9 +69,9 @@ FeatureManager::getlabel(map<string, map<string, int> > qrel2, const Document &d
 }
 
 Xapian::RankList
-FeatureManager::create_rank_list(const Xapian::MSet & mset, std::string & qid)
+FeatureManager::create_rank_list(const Xapian::MSet & mset, std::string & qid, int TrainorTest)
```

Comment:
Then shouldn't it be a `bool` argument? And if it's boolean, you don't need to call it `train_or_test`, you can just call it `train` (and explain the distinction in the docstring).